app/api/business/supplier_business.py

```diff
@@ -1,8 +1,32 @@
-from app.api.business.validators.supplier_validator import SupplierValidator
-from app.api.services import suppliers
+from app.api.business.validators import SupplierValidator
+from app.api.services import suppliers, application_service
 
 
-def get_supplier_validation_result(code):
+def get_supplier_messages(code):
     supplier = suppliers.get_supplier_by_code(code)
-    validation_result = SupplierValidator(supplier).validate_all()
-    return validation_result
+    result = SupplierValidator(supplier).validate_all()
+    application = application_service.find(
+        supplier_code=code,
+        status='submitted'
+    ).one_or_none()
+
+    if application:
+        return None
+
+    for cs in supplier.case_studies:
+        if cs.status == 'rejected':
+            result.warnings.append({
+                'message': 'Case study "{}" has been rejected'.format(cs.data.get('title')),
```

Comment:
Is there enough information here for a seller to know what to do next?

Reply:
awaiting content